realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,423 @@
+    @Test
+    public void setManagedLinkingObjectsThrows() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent = realm.createObject(AllJavaTypes.class, 2);
+        parent.setFieldObject(child);
+
+        try {
+            // Trying to set @LinkingObjects on a managed object is illegal
+            parent.setObjectParents(realm.where(AllJavaTypes.class).findAll());
+            fail("Should not be able to set a backlinked field");
+        } catch (UnsupportedOperationException ignored) {
+        } finally {
+            realm.cancelTransaction();
+        }
+    }
+
+    // Setting the linked object field creates the correct backlink
+    @Test
+    public void singleBacklink_link() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent = realm.createObject(AllJavaTypes.class, 2);
+        parent.setFieldObject(child);
+        realm.commitTransaction();
+
+        assertEquals(1, child.getObjectParents().size());
+        assertTrue(child.getObjectParents().contains(parent));
+    }
+
+    // Setting a linked list field creates the correct backlink
+    @Test
+    public void singleBacklink_linkList() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent = realm.createObject(AllJavaTypes.class, 2);
+        parent.getFieldList().add(child);
+        realm.commitTransaction();
+
+        assertEquals(1, child.getListParents().size());
+        assertTrue(child.getListParents().contains(parent));
+    }
+
+    // Setting multiple object links creates multiple backlinks
+    @Test
+    public void multipleBacklinks_link() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent1 = realm.createObject(AllJavaTypes.class, 2);
+        AllJavaTypes parent2 = realm.createObject(AllJavaTypes.class, 3);
+        parent1.setFieldObject(child);
+        parent2.setFieldObject(child);
+        realm.commitTransaction();
+
+        assertEquals(2, child.getObjectParents().size());
+    }
+
+    // Setting multiple list links creates multiple backlinks
+    @Test
+    public void multipleBacklinks_linkList() {
+        realm.beginTransaction();
+        AllJavaTypes child = realm.createObject(AllJavaTypes.class, 1);
+        AllJavaTypes parent1 = realm.createObject(AllJavaTypes.class, 2);
+        AllJavaTypes parent2 = realm.createObject(AllJavaTypes.class, 3);
+        parent1.getFieldList().add(child);
+        parent2.getFieldList().add(child);
+        realm.commitTransaction();
+
+        assertEquals(2, child.getListParents().size());
+    }
+
+    // Adding multiple list links creates multiple backlinks,
+    // even if the links are to a single object
+    // !!!FIXME: DOCUMENT THIS
```

Comment:
Well if I recall right, then any object link from linking objects to the target object are seen by the target as a `RealmResults`, right? So it makes sense, and I guess this means that each `RealmList` field has its own corresponding "backlink" list of linking objects, which also makes sense

Documenting it probably helps people understand how it works.